app/src/main/java/com/amaze/filemanager/ui/activities/MainActivity.java

```diff
@@ -580,6 +584,23 @@ private void initFabToSave(final ArrayList<Uri> uris) {
                       .show();
                   finish();
                 }
+              } else {
+                Bundle extras = intent.getExtras();
+                String data =
+                    extras.getString(Intent.EXTRA_SUBJECT)
+                        + "\nURL: "
+                        + extras.getString(Intent.EXTRA_TEXT);
+                String fileName = Long.toString(System.currentTimeMillis());
+                FileUtil.mktextfile(data, getCurrentMainFragment().getCurrentPath(), fileName);
```

Comment:
If you notice, the call is sync, so saving the data occurs in main thread, which is not recommended. 
Please write this call around in runnable as follows.
```AppConfig.runInBackground(() -> FileUtil.mktextfile(data, getCurrentMainFragment().getCurrentPath(), fileName));```